runtime/dynstats.c

```diff
@@ -624,3 +947,165 @@ dynstats_inc(dynstats_bucket_t *b, uchar* metric) {
 	RETiRet;
 }
 
+static struct wrkrInfo_s {
+	pthread_t tid;	/* the worker's thread ID */
+} wrkrInfo;
+
+typedef struct file_write_entry_s {
+	STAILQ_ENTRY(file_write_entry_s) link;
+	const char* name;
+	const char* content;
+	dynstats_bucket_t *bucket;
+} file_write_entry_t;
+
+typedef struct file_write_queue_s {
+	STAILQ_HEAD(head, file_write_entry_s) q;
+	STATSCOUNTER_DEF(ctrEnq, mutCtrEnq);
+	int size;
+	int ctrMaxSz;
+	statsobj_t *stats;
+	pthread_mutex_t mut;
+	pthread_cond_t wakeup_worker;
+} file_write_queue_t;
+
+static file_write_queue_t work_q;
+static int wrkrRunning;
+
+
+static void
+processWorkItem(file_write_entry_t *task) {
+	// write out the file
+	DBGPRINTF("dynstats: worker processing - '%s', '%s'\n", task->name, task->content);
+	writeStateFile(task->bucket, task->name, task->content);
```

Comment:
`task->content` is the invalid access I talk about in context of `put(json)`.

Reply:
I think we're safe, as we actually make a copy of the content here: https://github.com/rsyslog/rsyslog/blob/74a1d45d26d4b1a8bcb90b62af8fc1ddc97086b5/runtime/dynstats.c#L1074

Reply:
Yes, makes sense. Sorry for the false alarm.